sdk/resourcemanager/Azure.ResourceManager/docs/CONTRIBUTING.md

```diff
@@ -34,23 +34,84 @@ We use [NUnit 3][nunit] as our testing framework.
+### Running tests
 
-### Recorded tests
+In order to run the tests, the following environment variables need to be set:
 
-Our testing framework supports recording service requests made during a unit test so they can be replayed later. You can set the `AZURE_TEST_MODE` environment variable to `Playback` to run previously recorded tests, `Record` to record or re-record tests, and `Live` to run tests against the live service.
+| Name | Value | Description |
+| :--- | :---- | :---------- |
+| AZURE_TEST_MODE | Record<sup>1</sup> | Specify in which mode the test will run |
+| AZURE_AUTHORITY_HOST | https://login.microsoftonline.com<sup>2</sup> | The host of the Azure Active Directory authority |
+| AZURE_CLIENT_ID | TBD<sup>3</sup> | The Service Principal Application ID |
+| AZURE_CLIENT_SECRET | TBD<sup>3</sup> | A Service Principal Authentication Key |
+| AZURE_SUBSCRIPTION_ID | TBD<sup>3</sup> | The Azure Subscription ID |
+| AZURE_TENANT_ID | TBD<sup>3</sup> | The AAD Tenant ID |
 
-Properly supporting recorded tests does require a few extra considerations. All random values should be obtained via `this.Recording.Random` since we use the same seed on test playback to ensure our client code generates the same "random" values each time. You can't share any state between tests or rely on ordering because you don't know the order they'll be recorded or replayed. Any sensitive values are redacted via the [`ConfigurationRecordedTestSanitizer`][tests_sanitized].
+**Note**:
 
-### Running tests
+1. Our testing framework supports three different test modes: `Live`, `Playback`, `Record`. In management plane, please set the `AZURE_TEST_MODE` to `Record` for your first test run, this will record HTTP requests and responses and store the record files in `SessionRecords` folder. Properly supporting recorded tests does require a few extra considerations. All random values should be obtained via `this.Recording.Random` since we use the same seed on test playback to ensure our client code generates the same "random" values each time. You can't share any state between tests or rely on ordering because you don't know the order they'll be recorded or replayed. Any sensitive values are redacted via the [`ConfigurationRecordedTestSanitizer`](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/core/Azure.Core.TestFramework#sanitizing). After you have successfully recorded all the tests for the first time, you can change its value to `Playback`. If the tests locally fail due to recording session file mismatches at this point, the attribute `RecordedTest` will help enable automatically re-record failed tests.
```

Comment:
Yes, the default model is `playback`, you can see the code [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L581). And I agree with your description about the different behavior of the test mode in .Net, it seems that we have some differences about the test mode definitions between different languages. And I didn't mention the `live` test mode here because in mgmt plane:

- Locally we just need to run the test in record mode, which is more like a superset of live test. 
- In CI, we hardly ever run live tests because they cost too much time and maybe also too much money like `EventHub`.